### PR TITLE
New tasklists also allow shorthand syntax

### PR DESCRIPTION
**Pull request by HonkingGoose**
### Why:

@omerbensaadon is a PM with GitHub, they said that the new tasklist feature also accepts `#45` shorthand-syntax. The current docs say you _must_ use the full URL like this: `https://github.com/octo-org/octo-repo/issues/45`

Fixes a part of issue #24862. It should not close the issue as more work is needed after my PR gets merged. 😉 

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Rewrite a list item to allow both full URL and shorthand syntax, with examples of both.

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by cmwilson21** [comment]
👋 Hey @HonkingGoose - great to have another contribution from you 🎉 

I'll get this triaged for review! :zap:

**Comment by HonkingGoose** [comment]
While you guys are triaging my change, also check if _other_ parts of the document need to be changed as well. For example there's an example that only shows the full URL, and no shorthand-syntax:

> Your finished tasklist should look like this:
> 
> ````
> ```[tasklist]
> ### My tasks
> - [ ] https://github.com/octo-org/octo-repo/issues/45
> - [ ] Draft issue title
> ```
> ````

I could add a line like this to the `[tasklist]` above, to explain that shorthand-syntax is also allowed:

```
- [ ] #45 Shorthand-syntax is also allowed
```

Also I don't know if `shorthand-syntax` is the correct term for the "short URLs". 😄

File: content/issues/tracking-your-work-with-issues/about-tasklists.md
Comment (omerbensaadon) [inline content/issues/tracking-your-work-with-issues/about-tasklists.md:48]:
```suggestion
   - You may provide a full link to an issue or pull request, or use the shorthand `#ISSUE_NO` syntax. For example, `https://github.com/octo-org/octo-repo/issues/45` or `#45`.
```

Maybe this is clearer? 🤔 @stevecat defer to you here

**Comment by omerbensaadon** [comment]
Thank you for your contribution @HonkingGoose

File: content/issues/tracking-your-work-with-issues/about-tasklists.md
Comment (stevecat) [reply content/issues/tracking-your-work-with-issues/about-tasklists.md:48]: I really like including this! I made one further change to your suggestion, Omer, to make it clear that everything needs to be in the same repo in order to use the shorthand syntax.

```suggestion
   - You may provide a full link to an issue or pull request or, if the issue or pull request is in the same repository as your tasklist, you can use the #ISSUE-NUMBER shorthand syntax. For example, `https://github.com/octo-org/octo-repo/issues/45` or `#45`.
```

I'll apply this, approve the PR, and get this merged.

**Review by guntrip — APPROVED** [review]
Thanks so much for adding this, @HonkingGoose! I'll get this merged shortly and make sure we include this in future iterations of these docs.

File: content/issues/tracking-your-work-with-issues/about-tasklists.md
Comment (HonkingGoose) [inline content/issues/tracking-your-work-with-issues/about-tasklists.md:51]: @stevecat before you merge, did you check if you want to update the `[tasklist]` example on line 55 through 61? Maybe that example should also show the shorthand-syntax?

Comment (stevecat) [reply content/issues/tracking-your-work-with-issues/about-tasklists.md:51]: @HonkingGoose Thanks for the reminder! ✨ I think we'll keep this as the optional step in the procedural for now. I think the example is super clear at the moment.

Comment (HonkingGoose) [reply content/issues/tracking-your-work-with-issues/about-tasklists.md:51]: That's okay! Thanks for merging my contribution! 😄

**Pull request merged**
